cbor/shared/src/main/scala/fs2/data/cbor/high/internal/ValueParser.scala

```diff
@@ -209,16 +208,31 @@ object ValueParser {
   private def parseTags[F[_]](chunk: Chunk[CborItem],
                               idx: Int,
                               rest: Stream[F, CborItem],
-                              tags: CborValue => CborValue,
+                              tags: CborValue => Pull[F, Nothing, CborValue],
                               chunkAcc: List[CborValue])(implicit
-      F: RaiseThrowable[F]): Pull[F, CborValue, Result[F, CborValue => CborValue]] =
+      F: RaiseThrowable[F]): Pull[F, CborValue, Result[F, CborValue => Pull[F, Nothing, CborValue]]] =
     if (idx >= chunk.size) {
       Pull.output(Chunk.seq(chunkAcc.reverse)) >> rest.pull.uncons.flatMap {
         case Some((hd, tl)) => parseTags(hd, 0, tl, tags, Nil)
         case None           => Pull.raiseError(new CborParsingException("unexpected end of input"))
       }
     } else {
       chunk(idx) match {
+        case CborItem.Tag(Tags.PositiveBigNum) =>
+          def f: CborValue => Pull[F, Nothing, CborValue] = {
+            case CborValue.ByteString(bytes) => tags(CborValue.Integer(BigInt(bytes.toArrayUnsafe)))
+            case CborValue.Tagged(tag2, v)   => f(v).map(CborValue.Tagged(tag2, _))
```

Comment:
I think this is not correct. The specification states:

> A tag applies semantics to the data item it encloses. Tags can nest: if tag A encloses tag B, which encloses data item C, tag A applies to the result of applying tag B on data item C.

So `f` should apply to the decoding of value tagged by `tag2`.

Reply:
Should be fixed now.